fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java

```diff
@@ -240,20 +259,32 @@ public void markGroupStable(GroupId groupId, boolean needEditLog) {
         }
     }
 
-    public boolean removeTable(long tableId) {
+    public boolean removeTable(long tableId, OlapTable tbl, boolean isReplay) {
         writeLock();
         try {
             if (!table2Group.containsKey(tableId)) {
                 return false;
             }
 
             GroupId groupId = table2Group.remove(tableId);
+
+            if (tbl != null && tbl.isLakeTable() && !isReplay) {
+                List<Long> shardGroupIds = tbl.getShardGroupIds();
+                try {
+                    GlobalStateMgr.getCurrentStarOSAgent().updateMetaGroup(groupId.grpId, shardGroupIds,
+                            false /* isJoin */);
```

Comment:
When to delete the meta group?

Reply:
update meta group already supported delete option, but not in current 0.2.1 star manager version, will submit another pr when 0.2.2 is released.

Reply:
when last table reference is removed, meta group will be removed at the same time